In [1]:
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/gdrive')

Mounted at /gdrive


In [3]:
import os

In [4]:
dossier = '/gdrive/My Drive/DataKev/data/'
os.listdir(dossier)

['movie.csv']

In [5]:
movie = pd.read_csv(dossier + 'movie.csv' )

In [7]:
movie.head()

color      director_name  ...  aspect_ratio  movie_facebook_likes
0  Color      James Cameron  ...          1.78                 33000
1  Color     Gore Verbinski  ...          2.35                     0
2  Color         Sam Mendes  ...          2.35                 85000
3  Color  Christopher Nolan  ...          2.35                164000
4    NaN        Doug Walker  ...           NaN                     0

[5 rows x 28 columns]

In [8]:
movie.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

In [10]:
type(movie.columns)

pandas.core.indexes.base.Index

In [11]:
movie.indexWorking with Time Series

RangeIndex(start=0, stop=4916, step=1)

In [12]:
movie.values

array([['Color', 'James Cameron', 723.0, ..., 7.9, 1.78, 33000],
       ['Color', 'Gore Verbinski', 302.0, ..., 7.1, 2.35, 0],
       ['Color', 'Sam Mendes', 602.0, ..., 6.8, 2.35, 85000],
       ...,
       ['Color', 'Benjamin Roberds', 13.0, ..., 6.3, nan, 16],
       ['Color', 'Daniel Hsia', 14.0, ..., 6.3, 2.35, 660],
       ['Color', 'Jon Gunn', 43.0, ..., 6.6, 1.85, 456]], dtype=object)

In [ ]:
#nan : NOt a number

In [13]:
movie.values.shape

(4916, 28)

In [18]:
movie.head(2)

color   director_name  ...  aspect_ratio  movie_facebook_likes
0  Color   James Cameron  ...          1.78                 33000
1  Color  Gore Verbinski  ...          2.35                     0

[2 rows x 28 columns]

In [19]:
movie.tail()

color     director_name  ...  aspect_ratio  movie_facebook_likes
4911  Color       Scott Smith  ...           NaN                    84
4912  Color               NaN  ...         16.00                 32000
4913  Color  Benjamin Roberds  ...           NaN                    16
4914  Color       Daniel Hsia  ...          2.35                   660
4915  Color          Jon Gunn  ...          1.85                   456

[5 rows x 28 columns]

In [20]:
movie.sample(5)

color     director_name  ...  aspect_ratio  movie_facebook_likes
4065  Color     Alexandre Aja  ...          2.35                     0
1257  Color   Russell Mulcahy  ...          1.85                     0
4406  Color  Adam Jay Epstein  ...          1.85                   636
3265  Color      Peter Faiman  ...          2.35                     0
654   Color     Todd Phillips  ...          2.35                 25000

[5 rows x 28 columns]

In [21]:
movie.dtypes

color                         object
director_name                 object
num_critic_for_reviews       float64
duration                     float64
director_facebook_likes      float64
actor_3_facebook_likes       float64
actor_2_name                  object
actor_1_facebook_likes       float64
gross                        float64
genres                        object
actor_1_name                  object
movie_title                   object
num_voted_users                int64
cast_total_facebook_likes      int64
actor_3_name                  object
facenumber_in_poster         float64
plot_keywords                 object
movie_imdb_link               object
num_user_for_reviews         float64
language                      object
country                       object
content_rating                object
budget                       float64
title_year                   float64
actor_2_facebook_likes       float64
imdb_score                   float64
aspect_ratio                 float64
m

In [22]:
movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4916 entries, 0 to 4915
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      4897 non-null   object 
 1   director_name              4814 non-null   object 
 2   num_critic_for_reviews     4867 non-null   float64
 3   duration                   4901 non-null   float64
 4   director_facebook_likes    4814 non-null   float64
 5   actor_3_facebook_likes     4893 non-null   float64
 6   actor_2_name               4903 non-null   object 
 7   actor_1_facebook_likes     4909 non-null   float64
 8   gross                      4054 non-null   float64
 9   genres                     4916 non-null   object 
 10  actor_1_name               4909 non-null   object 
 11  movie_title                4916 non-null   object 
 12  num_voted_users            4916 non-null   int64  
 13  cast_total_facebook_likes  4916 non-null   int64

In [24]:
movie.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

In [25]:
movie.director_name

0           James Cameron
1          Gore Verbinski
2              Sam Mendes
3       Christopher Nolan
4             Doug Walker
              ...        
4911          Scott Smith
4912                  NaN
4913     Benjamin Roberds
4914          Daniel Hsia
4915             Jon Gunn
Name: director_name, Length: 4916, dtype: object

In [26]:
movie['director_name']

0           James Cameron
1          Gore Verbinski
2              Sam Mendes
3       Christopher Nolan
4             Doug Walker
              ...        
4911          Scott Smith
4912                  NaN
4913     Benjamin Roberds
4914          Daniel Hsia
4915             Jon Gunn
Name: director_name, Length: 4916, dtype: object

In [27]:
type(movie['director_name'])

pandas.core.series.Series

In [28]:
type(movie)

pandas.core.frame.DataFrame

In [29]:
movie.shape

(4916, 28)

In [31]:
movie['director_name'].shape

(4916,)

In [33]:
movie.head(2)

color   director_name  ...  aspect_ratio  movie_facebook_likes
0  Color   James Cameron  ...          1.78                 33000
1  Color  Gore Verbinski  ...          2.35                     0

[2 rows x 28 columns]

In [35]:
movie[['genres','director_name', 'duration']]

genres      director_name  duration
0     Action|Adventure|Fantasy|Sci-Fi      James Cameron     178.0
1            Action|Adventure|Fantasy     Gore Verbinski     169.0
2           Action|Adventure|Thriller         Sam Mendes     148.0
3                     Action|Thriller  Christopher Nolan     164.0
4                         Documentary        Doug Walker       NaN
...                               ...                ...       ...
4911                     Comedy|Drama        Scott Smith      87.0
4912     Crime|Drama|Mystery|Thriller                NaN      43.0
4913            Drama|Horror|Thriller   Benjamin Roberds      76.0
4914             Comedy|Drama|Romance        Daniel Hsia     100.0
4915                      Documentary           Jon Gunn      90.0

[4916 rows x 3 columns]

In [37]:
movie.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

In [38]:
movie.loc[ :, 'director_name' ]

0           James Cameron
1          Gore Verbinski
2              Sam Mendes
3       Christopher Nolan
4             Doug Walker
              ...        
4911          Scott Smith
4912                  NaN
4913     Benjamin Roberds
4914          Daniel Hsia
4915             Jon Gunn
Name: director_name, Length: 4916, dtype: object

In [39]:
movie.loc[ :, ['genres','director_name', 'duration'] ]

genres      director_name  duration
0     Action|Adventure|Fantasy|Sci-Fi      James Cameron     178.0
1            Action|Adventure|Fantasy     Gore Verbinski     169.0
2           Action|Adventure|Thriller         Sam Mendes     148.0
3                     Action|Thriller  Christopher Nolan     164.0
4                         Documentary        Doug Walker       NaN
...                               ...                ...       ...
4911                     Comedy|Drama        Scott Smith      87.0
4912     Crime|Drama|Mystery|Thriller                NaN      43.0
4913            Drama|Horror|Thriller   Benjamin Roberds      76.0
4914             Comedy|Drama|Romance        Daniel Hsia     100.0
4915                      Documentary           Jon Gunn      90.0

[4916 rows x 3 columns]

In [40]:
movie.index

RangeIndex(start=0, stop=4916, step=1)

In [42]:
movie.loc[2: 5, 'director_name' ]

2           Sam Mendes
3    Christopher Nolan
4          Doug Walker
5       Andrew Stanton
Name: director_name, dtype: object

In [43]:
movie.loc[[6, 4, 100], 'director_name' ]

6        Sam Raimi
4      Doug Walker
100      Rob Cohen
Name: director_name, dtype: object

In [44]:
movie.loc[[6, 4, 100], 'director_name': 'duration' ]

director_name  num_critic_for_reviews  duration
6       Sam Raimi                   392.0     156.0
4     Doug Walker                     NaN       NaN
100     Rob Cohen                   187.0     106.0

In [45]:
movie.iloc[[6, 4, 100], 5:9 ]

actor_3_facebook_likes  actor_2_name  actor_1_facebook_likes        gross
6                    4000.0  James Franco                 24000.0  336530303.0
4                       NaN    Rob Walker                   131.0          NaN
100                  4000.0    Vin Diesel                 23000.0  144512310.0

In [46]:
movie.describe()

num_critic_for_reviews     duration  ...  aspect_ratio  movie_facebook_likes
count             4867.000000  4901.000000  ...   4590.000000           4916.000000
mean               137.988905   107.090798  ...      2.222349           7348.294142
std                120.239379    25.286015  ...      1.402940          19206.016458
min                  1.000000     7.000000  ...      1.180000              0.000000
25%                 49.000000    93.000000  ...      1.850000              0.000000
50%                108.000000   103.000000  ...      2.350000            159.000000
75%                191.000000   118.000000  ...      2.350000           2000.000000
max                813.000000   511.000000  ...     16.000000         349000.000000

[8 rows x 16 columns]

In [47]:
movie['director_name'].describe()

count                 4814
unique                2397
top       Steven Spielberg
freq                    26
Name: director_name, dtype: object

Dans notre dataset, nous avons, 2397 directeurs uniques avec Steven Spielberg étant celui avec le plus de film qui est égale à 26

In [49]:
movie['director_name'].value_counts()

Steven Spielberg    26
Woody Allen         22
Martin Scorsese     20
Clint Eastwood      20
Ridley Scott        16
                    ..
Harmony Korine       1
Inna Evlannikova     1
Hsiao-Hsien Hou      1
Christine Jeffs      1
Vincent Ward         1
Name: director_name, Length: 2397, dtype: int64

In [50]:
movie.num_critic_for_reviews.describe()

count    4867.000000
mean      137.988905
std       120.239379
min         1.000000
25%        49.000000
50%       108.000000
75%       191.000000
max       813.000000
Name: num_critic_for_reviews, dtype: float64

In [51]:
movie.num_critic_for_reviews.std()

120.23937916791036

In [52]:
movie.mean(axis=0)

num_critic_for_reviews       1.379889e+02
duration                     1.070908e+02
director_facebook_likes      6.910145e+02
actor_3_facebook_likes       6.312763e+02
actor_1_facebook_likes       6.494488e+03
gross                        4.764451e+07
num_voted_users              8.264492e+04
cast_total_facebook_likes    9.579816e+03
facenumber_in_poster         1.377320e+00
num_user_for_reviews         2.676688e+02
budget                       3.654749e+07
title_year                   2.002448e+03
actor_2_facebook_likes       1.621924e+03
imdb_score                   6.437429e+00
aspect_ratio                 2.222349e+00
movie_facebook_likes         7.348294e+03
dtype: float64

In [53]:
movie.sum(axis=0)

num_critic_for_reviews                                                  671592
duration                                                                524852
director_facebook_likes                                            3.32654e+06
actor_3_facebook_likes                                             3.08884e+06
actor_1_facebook_likes                                             3.18814e+07
gross                                                              1.93151e+11
genres                       Action|Adventure|Fantasy|Sci-FiAction|Adventur...
movie_title                  AvatarPirates of the Caribbean: At World's End...
num_voted_users                                                      406282451
cast_total_facebook_likes                                             47094375
facenumber_in_poster                                                      6753
movie_imdb_link              http://www.imdb.com/title/tt0499549/?ref_=fn_t...
num_user_for_reviews                                

In [56]:
genre_un = movie.genres[0]
genre_un

'Action|Adventure|Fantasy|Sci-Fi'

In [57]:
genre_un.split('|')

['Action', 'Adventure', 'Fantasy', 'Sci-Fi']

In [58]:
movie['genres']

0       Action|Adventure|Fantasy|Sci-Fi
1              Action|Adventure|Fantasy
2             Action|Adventure|Thriller
3                       Action|Thriller
4                           Documentary
                     ...               
4911                       Comedy|Drama
4912       Crime|Drama|Mystery|Thriller
4913              Drama|Horror|Thriller
4914               Comedy|Drama|Romance
4915                        Documentary
Name: genres, Length: 4916, dtype: object

In [60]:
movie.values

array([['Color', 'James Cameron', 723.0, ..., 7.9, 1.78, 33000],
       ['Color', 'Gore Verbinski', 302.0, ..., 7.1, 2.35, 0],
       ['Color', 'Sam Mendes', 602.0, ..., 6.8, 2.35, 85000],
       ...,
       ['Color', 'Benjamin Roberds', 13.0, ..., 6.3, nan, 16],
       ['Color', 'Daniel Hsia', 14.0, ..., 6.3, 2.35, 660],
       ['Color', 'Jon Gunn', 43.0, ..., 6.6, 1.85, 456]], dtype=object)